src/framework/handlers/animation.js

```diff
@@ -52,31 +52,35 @@ class AnimationHandler {
             }
         }
 
-        http.get(url.load, options, function (err, response) {
+        http.get(url.load, options, (err, response) => {
             if (err) {
                 callback(`Error loading animation resource: ${url.original} [${err}]`);
             } else {
-                callback(null, response);
+                // parse the result immediately (this used to happen during open)
+                if (path.getExtension(url.original).toLowerCase() === '.glb') {
+                    GlbParser.parse('filename.glb', response, null, null, (err, parseResult) => {
+                        if (err) {
+                            callback(err);
+                        } else {
+                            const animations = parseResult.animations;
+                            if (asset?.data?.events) {
```

Comment:
unrelated to this PR, but this feels strange @ellthompson - we have a single asset.data.events, but create multiple instances of AnimEvents and attach it to all animations. Is this correct?